plugins/uefi-dbx/fu-self-test.c

```diff
@@ -45,17 +51,326 @@ fu_efi_image_func(void)
+
+	if (tc->with_snapd && !fixture->mock_snapd_available) {
+		g_test_skip("mock snapd not available");
+		return;
+	}
+
+	ret = fu_context_load_quirks(ctx,
+				     FU_QUIRKS_LOAD_FLAG_NO_CACHE | FU_QUIRKS_LOAD_FLAG_NO_VERIFY,
+				     &error);
+	g_assert_no_error(error);
+	g_assert_true(ret);
+
+	ret = fu_plugin_runner_startup(plugin, progress, &error);
+	g_assert_no_error(error);
+	g_assert_true(ret);
+
+	if (tc->with_snapd) {
+		FuUefiDbxSnapdNotifier *notifier = NULL;
+		gboolean supported = FALSE;
+
+		supported = fu_uefi_dbx_self_test_plugin_get_snapd_integration_supported(
+		    FU_UEFI_DBX_PLUGIN(plugin));
+		g_assert_cmpint(tc->snapd_supported, ==, supported);
+
+		notifier =
+		    fu_uefi_dbx_self_test_plugin_get_snapd_notifier(FU_UEFI_DBX_PLUGIN(plugin));
+		if (tc->snapd_supported) {
+			g_assert_nonnull(notifier);
+			g_object_unref(notifier);
+		} else {
+			g_assert_null(notifier);
+		}
+	}
+}
+
 int
 main(int argc, char **argv)
 {
+	g_autofree gchar *testdatadir = NULL;
+	FuTestCase simple = {
+	    .with_snapd = FALSE,
+	    .devices_cnt = 0,
+	};
+	FuTestCase with_snapd = {
+	    .with_snapd = TRUE,
+	    .snapd_supported = TRUE,
+	    .mock_snapd_scenario = "happy-startup",
+	};
+	FuTestCase with_snapd_no_support = {
+	    .with_snapd = TRUE,
+	    .snapd_supported = FALSE,
+	    .mock_snapd_scenario = "not-supported",
+	};
+
 	(void)g_setenv("G_TEST_SRCDIR", SRCDIR, FALSE);
 	g_test_init(&argc, &argv, NULL);
 
 	/* only critical and error are fatal */
 	g_log_set_fatal_mask(NULL, G_LOG_LEVEL_ERROR | G_LOG_LEVEL_CRITICAL);
 	(void)g_setenv("G_MESSAGES_DEBUG", "all", TRUE);
 
+	testdatadir = g_test_build_filename(G_TEST_DIST, "tests", NULL);
+
+	(void)g_setenv("FWUPD_SYSFSFWDIR", testdatadir, TRUE);
+	(void)g_setenv("FWUPD_EFIVARS", "dummy", TRUE);
+	(void)g_setenv("FWUPD_SYSFSDRIVERDIR", testdatadir, TRUE);
+	(void)g_setenv("FWUPD_SYSFSFWATTRIBDIR", testdatadir, TRUE);
+
 	/* tests go here */
 	g_test_add_func("/uefi-dbx/image", fu_efi_image_func);
+
+	g_test_add("/uefi-dbx/startup",
+		   FuTestFixture,
+		   &simple,
+		   fu_self_test_set_up,
+		   fu_uefi_dbx_test_plugin_startup,
+		   fu_self_test_tear_down);
+	g_test_add("/uefi-dbx/startup/snapd/supported",
+		   FuTestFixture,
+		   &with_snapd,
+		   fu_self_test_set_up,
+		   fu_uefi_dbx_test_plugin_startup,
+		   fu_self_test_tear_down);
+	g_test_add("/uefi-dbx/startup/snapd/not-supported",
+		   FuTestFixture,
+		   &with_snapd_no_support,
+		   fu_self_test_set_up,
+		   fu_uefi_dbx_test_plugin_startup,
+		   fu_self_test_tear_down);
+
+	g_test_add("/uefi-dbx/coldplug/no-dev",
+		   FuTestFixture,
+		   &simple,
+		   fu_self_test_set_up,
+		   fu_uefi_dbx_test_plugin_coldplug_no_device,
+		   fu_self_test_tear_down);
+
+	g_test_add("/uefi-dbx/coldplug/with-dev",
+		   FuTestFixture,
+		   &simple,
+		   fu_self_test_set_up,
+		   fu_uefi_dbx_test_plugin_coldplug_probed_device,
+		   fu_self_test_tear_down);
+
```

Comment:
Some of the tests here could be proposed in a separate PR as they improve overall test coverage of the plugin. Let me know if that's preferred and I'll split out the changes.

Reply:
I think leaving them together is totally fine, they're great thanks!